flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/job/SubtaskExecutionAttemptDetailsInfo.java

```diff
@@ -237,6 +256,57 @@ public static SubtaskExecutionAttemptDetailsInfo create(
                 endTime,
                 duration,
                 ioMetricsInfo,
-                taskmanagerId);
+                taskmanagerId,
+                getExecutionStateDuration(execution));
+    }
+
+    private static Map<ExecutionState, Long> getExecutionStateDuration(AccessExecution execution) {
+        Map<ExecutionState, Long> executionStateDuration = new HashMap<>();
+        long now = System.currentTimeMillis();
+        ExecutionState state = execution.getState();
+        executionStateDuration.put(
+                ExecutionState.CREATED,
+                calculateStateDuration(
+                        execution.getStateTimestamp(ExecutionState.CREATED),
+                        execution.getStateEndTimestamp(ExecutionState.CREATED),
+                        state == ExecutionState.CREATED ? now : 0));
```

Comment:
```suggestion
        executionStateDuration.put(
                ExecutionState.CREATED,
                calculateStateDuration(
                        execution.getStateTimestamp(ExecutionState.CREATED),
                        state == ExecutionState.CREATED ? now : execution.getStateEndTimestamp(ExecutionState.CREATED)));
```

I think this helps prevent the implicity that indicating the current state with `now != 0`.